doc/releases/release-notes-2.4.rst

```diff
@@ -311,11 +324,72 @@ Drivers and Sensors
 
 * WiFi
 
+  * Add IPv6 support to Simplelink driver.
+  * Add DNS offloading support to eswifi driver.
+  * Fix esp driver offload protocol parsing.
+  * Fix esp driver GPIO reset control logic.
+  * Fix eswifi driver offloading packet parsing.
 
 
 Networking
 **********
 
+* Enable new TCP stack by default. The legacy TCP stack is not yet removed and
+  can be used if needed.
+* Make network interface a kernel object. This allows better access control
+  handling when usermode is enabled.
+* Use kernel stacks in network related threads, to save memory when usermode
+  is enabled.
+* Enable network statistics collection in key points of the network stack. This
+  can be used to get information where time is spent in RX or TX.
+* The BSD socket sendmsg() can now be used with AF_PACKET type sockets.
+* Add support for enabling OpenThread reference device.
+* Add support for enabling MQTT persistent sessions.
```

Comment:
Not much happened in area of MQTT, but one bug that caused double CONNACK event, in
case the server rejected the connection was fixed: https://github.com/zephyrproject-rtos/zephyr/pull/26967/commits/102f2591de53547132923bb837625e3e7aef8987

Reply:
> Not much happened in area of MQTT, but one bug that caused double CONNACK event, in
> case the server rejected the connection was fixed: [102f259](https://github.com/zephyrproject-rtos/zephyr/commit/102f2591de53547132923bb837625e3e7aef8987)

I did not mention all of the bug fixes but can add this if you think it is important.

Reply:
It affects what events the applcation is receiveing, so I think it's worth mentioning.